### PR TITLE
feat: adding register implementation and some changes in .gitignore

### PR DESCRIPTION
**Pull request by SilviaPabon**
- I add and update register implementation in ServiceImp, I didn't knew if I should verify some things like the credentials information is in a correct type
- Also, I update .gitignore
- Also I have a question, when ./gradlew run is execute it, there is a way that when we make changes we should't have to execute it again?

### CODE REVIEW

File: app/src/main/java/gateway/soap/ServiceImp.java [hunk 1]
```diff
@@ -17,9 +24,37 @@
...
+									  .uri (URI.create (url))
+									  .header ("Content-Type", "application/json")
+									  .build ();
+
+			HttpResponse<String> response =
+				client.send (request, HttpResponse.BodyHandlers.ofString ());
+
+			auth.token = response.body ();
+		} catch (IOException | InterruptedException e) {
+			e.printStackTrace ();
```
Comment (Woynert) [inline app/src/main/java/gateway/soap/ServiceImp.java:54]: On error, fill the appropiate fields `success` and `message` from [StatusRes](https://github.com/hawks-atlanta/gateway-java/blob/dev/app/src/main/java/gateway/soap/response/StatusRes.java) which [SessionRes](https://github.com/hawks-atlanta/gateway-java/blob/dev/app/src/main/java/gateway/soap/response/SessionRes.java) inherits.

**Review by Woynert — CHANGES_REQUESTED** [review]
Before fixing the issues mentioned get this branch up to date with `dev`.

The following changes where applied: [Add JSON library](https://github.com/hawks-atlanta/gateway-java/pull/37), and [Normalize Status Response](https://github.com/hawks-atlanta/gateway-java/pull/36).

File: app/src/main/java/gateway/soap/ServiceImp.java [hunk 2]
```diff
@@ -17,9 +24,37 @@
...
 	}
 
+	// user register in auth service, returns an access token
 	@WebMethod public Authorization register(Credentials credentials)
 	{
-		return new Authorization ();
+
+		Authorization auth = new Authorization ();
+
+		String url = EnvironmentVar.AUTH_BASEURL + "register";
```
[the remark this answers is not in the document]
Comment (SilviaPabon) [reply app/src/main/java/gateway/soap/ServiceImp.java:50]: @Woynert I saw you were working on something about it. I think I'm going to leave this alone until this is modified. AUTH_BASEURL had the host and port of the authentication service, in the other methods the others had to associate a base url as needed, that was my plan, but you can tell me what is most convenient to leave here. I will working on the other thing meanwhile.

**Review by Woynert — APPROVED** [review]
:hammer: Good work!
Squash and merge

**Pull request merged**